### PR TITLE
add mi dehumidifier v1 (nwt.derh.wdh318efw1)

### PR DESCRIPTION
**Pull request by stkang**
#Add support for WIDETECH WDH318EFW1 dehumidifier (nwt.derh.wdh318efw1) #534

### CODE REVIEW

**Comment by syssi** [comment]
Good job! Do you like to provide some tests, too?

**Comment by stkang** [comment]
@syssi right?

**Comment by stkang** [comment]
@syssi right?

**Comment by stkang** [comment]
please confirm and merge. :)

File: miio/tests/test_airdehumidifier.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        self.device.off()
+        assert self.is_on() is False
+
+    def test_status(self):
+        self.device._reset_state()
+
+        device_info = DeviceInfo(self.device.dummy_device_info)
+
+        assert repr(self.state()) ==\
+               repr(AirDehumidifierStatus(self.device.start_state, device_info))
```
[the remark this answers is not in the document]
Comment (stkang) [reply miio/tests/test_airdehumidifier.py]: modify

File: miio/tests/test_airdehumidifier.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        assert self.state().humidity == self.device.start_state["humidity"]
+        assert self.state().mode == OperationMode(self.device.start_state["mode"])
+        assert self.state().led == self.device.start_state["led"]
+        assert self.state().buzzer == (self.device.start_state["buzzer"] == 'on')
+        assert self.state().child_lock == (self.device.start_state["child_lock"] == 'on')
+        assert self.state().target_humidity == self.device.start_state["auto"]
+        assert self.state().fan_speed == FanSpeed(self.device.start_state["fan_speed"])
+        assert self.state().tank_full == (self.device.start_state["tank_full"] == 'on')
+        assert self.state().compressor_status == \
+               (self.device.start_state["compressor_status"] == 'on')
```
[the remark this answers is not in the document]
Comment (stkang) [reply miio/tests/test_airdehumidifier.py]: modify

File: miio/tests/test_airdehumidifier.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+        self.device.off()
+        assert self.is_on() is False
+
+    def test_status(self):
+        self.device._reset_state()
+
+        device_info = DeviceInfo(self.device.dummy_device_info)
+
+        assert repr(self.state()) == repr(AirDehumidifierStatus(self.device.start_state, device_info))
```
[the remark this answers is not in the document]
Comment (stkang) [reply miio/tests/test_airdehumidifier.py]: modify

File: miio/tests/test_airdehumidifier.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        assert self.state().temperature == self.device.start_state["temp"]
+        assert self.state().humidity == self.device.start_state["humidity"]
+        assert self.state().mode == OperationMode(self.device.start_state["mode"])
+        assert self.state().led == self.device.start_state["led"]
+        assert self.state().buzzer == (self.device.start_state["buzzer"] == 'on')
+        assert self.state().child_lock == (self.device.start_state["child_lock"] == 'on')
+        assert self.state().target_humidity == self.device.start_state["auto"]
+        assert self.state().fan_speed == FanSpeed(self.device.start_state["fan_speed"])
+        assert self.state().tank_full == (self.device.start_state["tank_full"] == 'on')
+        assert self.state().compressor_status == (self.device.start_state["compressor_status"] == 'on')
```
[the remark this answers is not in the document]
Comment (stkang) [reply miio/tests/test_airdehumidifier.py]: modify

**Comment by stkang** [comment]
@rytilahti 
please, checkand merge. :)

**Comment by rytilahti** [comment]
From the surface it looks good to me, but I'll let @syssi to merge it when he sees it's fine :-)

**Comment by stkang** [comment]
@syssi Thanks

**Comment by syssi** [comment]
I will prepare a release soonish.

**Pull request merged**